File: src/dataset/cpp/binary_search_Exercism_thefullarcticfox/gemini/GeminiFlash_binary_search_v2.cpp
#include "binary_search.h"
#include <stdexcept>
#include <algorithm>

namespace binary_search {
	size_t	find(std::vector<int> data, int value) {
		if (data.empty())
			throw std::domain_error("empty vector");

		size_t	left = 0, right = data.size(); // Correct right boundary
		while (left < right) { // Use < instead of <=
			size_t	middle = left + (right - left) / 2;

			if (data[middle] == value) // Use [] instead of at()
				return middle;

			if (data[middle] < value)
				left = middle + 1;
			else
				right = middle; // Correct right boundary update
		}
		throw std::domain_error("not found");
	}
}  // namespace binary_search

File: src/dataset/go/binary_search_Exercism_go_drapala/gemini/GeminiFlash_binary_search_v2.go
package binarysearch

func SearchInts(list []int, key int) int {
	l := 0
	r := len(list)
	for l < r {
		m := l + (r-l)/2
		if list[m] < key {
			l = m + 1
		} else if list[m] > key {
			r = m
		} else {
			return m
		}
	}
	return -1
}